Sources/SwiftMemcache/MemcachedConnection.swift

```diff
@@ -0,0 +1,170 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-memcache-gsoc open source project
+//
+// Copyright (c) 2023 Apple Inc. and the swift-memcache-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-memcache-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+@_spi(AsyncChannel)
+
+import NIOCore
+import NIOPosix
+
+/// An actor to create a connection to a Memcache server.
+///
+/// This actor can be used to send commands to the server.
+public actor MemcachedConnection {
+    private typealias StreamElement = (MemcachedRequest, CheckedContinuation<MemcachedResponse, Error>)
+    /// The underlying channel to communicate with the server.
+    private let channel: NIOAsyncChannel<MemcachedResponse, MemcachedRequest>
+    /// The channel's event loop group.
+    private let eventLoopGroup: EventLoopGroup
```

Comment:
We might want to keep allocator in the running state - since its only really used in our `set` method currently within the `running` state?

Reply:
We can. I am actually thinking that we should get rid of the allocator here all together. Instead of allocating a byte buffer here and then later writing the byte buffer when we encode the request, we could just pass the value into the request by making the request generic over it. Then when we encode we could just write it into the final buffer. Though let's do this in a follow up PR.